### PR TITLE
Improve disableShadowRelocate

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/issues/9711

### CODE REVIEW

**Comment by 123liuziming** [comment]
I've tried to use a similar approach to fix the problem before, but some tests like `SpringWebfluxTest` failed.
And I 've tried your solution locally, the `SpringWebfluxTest` failed as well:
image

**Comment by laurit** [comment]
@123liuziming it is expected that some tests will fail with this flag. From your screenshot it is not possible to tell why webflux tests fail.

**Comment by 123liuziming** [comment]
> @123liuziming it is expected that some tests will fail with this flag. From your screenshot it is not possible to tell why webflux tests fail.

Sorry, I 

> @123liuziming it is expected that some tests will fail with this flag. From your screenshot it is not possible to tell why webflux tests fail.

Sorry, I didn't read your comment in issue carefully. Now it does pass at least part of the tests！Thanks！:thumbsup: :thumbsup: :thumbsup:

**Review by trask — APPROVED** [review]

**Review by mateuszrzeszutek — APPROVED** [review]

File: javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java [hunk 1]
```diff
@@ -164,8 +165,14 @@ private static boolean isJar(File f) {
 
   private static void addFileUrl(List<URL> result, File file) {
     try {
-      URL wrappedUrl = new URL("otel", null, -1, "/", new RemappingUrlStreamHandler(file));
-      result.add(wrappedUrl);
+      // skip shading extension classes if opentelemetry-api is not shaded (happens when using
+      // disableShadowRelocate=true)
+      if (Context.class.getName().contains("shaded")) {
```
Comment (trask) [inline javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/ExtensionClassLoader.java:170]: (mostly for readability)

```suggestion
      if (Context.class.getName().contains(".shaded.")) {
```

**Comment by laurit** [comment]
@123liuziming I made one more change that should also get the webflux tests passing

**Pull request merged**